R/reporter-list.R

```diff
@@ -31,32 +31,65 @@ ListReporter <- R6::R6Class("ListReporter",
     },
 
     add_result = function(context, test, result) {
-      if (!is.null(self$current_expectations)) {
-        self$current_expectations$push(result)
+      if (is.null(self$current_expectations)) {
+        # we received an unexpected result: could be a bare expectation or an exception/error
+        if (!inherits(result, 'error')) return()
```

Comment:
What's the point of this line?

Reply:
I guess the intent is to not capture bare expectations, just errors, but I'm not sure why that's desirable?

Reply:
Suppose that you have some outcode outside tests, e.g. to preprocess some test data. 
If that fails, you want your test script to crash so that you know there's something wrong and you can debug. 
Otherwise this could unnoticed, among the tests that failed.

Reply:
Oh I see what you mean — we need to make sure an error outside of a test case causes the code to terminate.

Reply:
Yes. And that probably should be handled somewhere, but not sure the ListReporter is the best place to do this, as this should probably be reporter-independent.

Reply:
Filed an https://github.com/r-lib/testthat/issues/950